index.js

```diff
@@ -117,15 +118,21 @@ module.exports = class Transaction {
    * @return {Buffer}
    */
   hash (signature) {
+    const rawCopy = this.raw.slice(0)
+    if (this._chainId) {
+      this.v = this._chainId
+      this.r = this.s = 0
+    }
     let toHash
 
-    if (typeof signature === 'undefined') {
+    if (this._chainId || typeof signature === 'undefined') {
       signature = true
     }
 
     toHash = signature ? this.raw : this.raw.slice(0, 6)
 
     // create hash
+    this.raw = rawCopy.slice(0)
```

Comment:
@axic the `this.v = this._chainId` changes the underlying raw data since this.v is a setter (i wouldn't be opposed to removing raw and this weird setter/getter stuff b/c its confusing).